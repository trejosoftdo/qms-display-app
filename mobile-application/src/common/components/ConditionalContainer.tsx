import * as React from 'react';

/**
 * ConditionalContainerProps defines the props for the Conditional Container Component.
 *
 * @property display - to indicate if the passed children are being displayed
 * @property children - the contained children
 */
interface ConditionalContainerProps {
  display?: boolean;
  children: React.ReactNode;
}

/**
 * A component that conditionally render a given set of children
 *
 * @param {ConditionalContainerProps} props - The props for the Conditional Container component.
 */
const ConditionalContainer: React.FC<ConditionalContainerProps> = (props: ConditionalContainerProps) => {
  if (!props.display) {
    return null;
  }
  return props.children;
}

export default ConditionalContainer;
