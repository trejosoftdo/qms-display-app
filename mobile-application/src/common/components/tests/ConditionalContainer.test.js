import React from 'react';
import ConditionalContainer from '../ConditionalContainer';


describe('ConditionalContainer component', () => {
  it('does not render the content if display is not active', () => {
    expect(renderShallow(
      <ConditionalContainer>
        <div>Test content</div>
      </ConditionalContainer>
    )).toMatchSnapshot();
  });

  it('does render the content if display is active', () => {
    expect(renderShallow(
      <ConditionalContainer display>
        <div>Test content</div>
      </ConditionalContainer>
    )).toMatchSnapshot();
  });
});