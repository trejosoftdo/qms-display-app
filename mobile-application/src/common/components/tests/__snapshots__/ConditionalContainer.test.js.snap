// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionalContainer component does not render the content if display is not active 1`] = `null`;

exports[`ConditionalContainer component does render the content if display is active 1`] = `
<div>
  Test content
</div>
`;
